lib/service/storage/elasticsearch.ts

```diff
@@ -842,20 +933,26 @@ class ElasticSearch extends Service {
    * @returns {Promise.<{ _id, _version }>}
    */
   async upsert(
-    index,
-    collection,
-    id,
-    content,
+    index: string,
+    collection: string,
+    id: string,
+    content: JSONObject,
     {
       defaultValues = {},
       refresh,
       userId = null,
       retryOnConflict,
       injectKuzzleMeta = true,
+    }: {
+      defaultValues?: JSONObject;
+      refresh?: boolean | "wait_for";
+      userId?: string;
+      retryOnConflict?: number;
+      injectKuzzleMeta?: boolean;
     } = {},
```

Comment:
```suggestion
    {
      defaultValues = {},
      refresh,
      userId = null,
      retryOnConflict,
      injectKuzzleMeta = true,
    }: Exclude<KRequestParams, "timeout"> & { defaultValues: JSONObject; } = {},
```

With new `KRequestParams` type we can compose type like it, to DRY